web/packages/teleterm/src/ui/services/workspacesService/workspacesService.ts

```diff
@@ -635,3 +649,20 @@ type UnifiedResourcePreferencesSchemaAsRequired = Required<
 export const useWorkspaceServiceState = () => {
   return useStoreSelector('workspacesService', identitySelector);
 };
+
+function getRestorableDocuments(documents: Document[]): Document[] {
+  return (
+    documents
+      // We don't restore 'doc.authorize_web_session' because the user would not
+      // be able to authorize a session at a later time anyway.
+      .filter(d => d.kind !== 'doc.authorize_web_session')
+  );
+}
+
+/** Assumes that there is at least one document to restore. */
+function getLocationToRestore(
+  documents: Document[],
+  location: DocumentUri
+): DocumentUri {
+  return documents.find(d => d.uri === location) ? location : documents[0]!.uri;
```

Comment:
Nit, but I think we should avoid TypeScript's `!` where possible. It's not much different than using `as any`. It's fine in rare cases where a 3rd-party library makes some guarantees that for some reason are not expressed by types.

But here in our code, if our heart was really in it, we'd use a non-empty array to guarantee that `documents` always have at least one element. I don't think people do it that often in TS because the ergonomics of it are much worse compared to doing the same in Elm or Haskell.

So in this particular scenario, I think it might be better to do `documents[0]?.uri` rather than keep our fingers crossed that this won't ever be empty. On line 425, we already made so that an empty array is used if `persistedWorkspace?.documents` doesn't return anything.

Reply:
Yeah, it's not really needed actually, it's fine for location to be `undefined`.